File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    std::array<int, 46> cache;
    int climbStairs(int n) {
        if(cache.size()){
            if(cache[n]>0){return cache[n];}
            if(n < 2){return 1;}
            // else if(n == 1){return 1;}
            else{cache[n] = climbStairs(n-1)+climbStairs(n-2);}
            
        // return 1;
        }
        return cache[n];
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        res_list = [0] * 2
        res_list[0] = 1
        res_list[1] = 1
        total = res_list[1]
        for i in range(2,n+1):
            total = res_list[0] + res_list[1]
            res_list[0],res_list[1] = res_list[1],total
        return total